engine/src/main/java/com/core/rpgmaker/AsarArchive.kt

```
@@ -0,0 +1,191 @@
+
+    fun read(path: String): ByteArray? {
+        return try {
+            val e = entries[normalize(path)] ?: return null
+            if (e.directory) return null
+            val data = ByteArray(e.size.toInt())
+            synchronized(raf) {
+                raf.seek(dataOffset + e.offset)
+                raf.readFully(data)
+            }
+            data
+        } catch (t: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)